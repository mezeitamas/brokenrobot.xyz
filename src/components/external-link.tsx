import React from 'react';
import type { FunctionComponent, PropsWithChildren } from 'react';

import { FiExternalLink } from 'react-icons/fi';

type ExternalLinkProps = {
    href: string;
};

const ExternalLink: FunctionComponent<PropsWithChildren<ExternalLinkProps>> = ({ href, children }) => {
    return (
        <div className="flex items-center gap-1 hover:underline">
            <a
                href={href}
                target="_blank"
            >
                {children}
            </a>
            <FiExternalLink />
        </div>
    );
};

export { ExternalLink };
