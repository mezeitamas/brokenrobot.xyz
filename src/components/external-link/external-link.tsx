import React from 'react';
import type { FunctionComponent, PropsWithChildren } from 'react';

import { FiExternalLink } from 'react-icons/fi';

type ExternalLinkProps = {
    href: string;
};

const ExternalLink: FunctionComponent<PropsWithChildren<ExternalLinkProps>> = ({ href, children }) => {
    return (
        <span className="inline-flex items-center gap-1 hover:underline">
            <a
                href={href}
                target="_blank"
            >
                {children}
            </a>
            <FiExternalLink size="14" />
        </span>
    );
};

export { ExternalLink };

export type { ExternalLinkProps };
