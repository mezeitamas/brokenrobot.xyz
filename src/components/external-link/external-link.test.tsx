import React from 'react';

import { create } from 'react-test-renderer';

import { ExternalLink } from './external-link';
import type { ExternalLinkProps } from './external-link';

describe('<ExternalLink />', () => {
    test('should render', () => {
        // Given
        const props: ExternalLinkProps = {
            href: 'https://www.brokenrobot.xyz'
        };
        const children = 'Broken Robot';

        // When
        const component = create(<ExternalLink {...props}>{children}</ExternalLink>).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
