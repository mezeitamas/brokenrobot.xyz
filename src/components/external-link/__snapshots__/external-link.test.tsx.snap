// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExternalLink /> should render 1`] = `
<span
  className="inline-flex items-center gap-1 hover:underline"
>
  <a
    href="https://www.brokenrobot.xyz"
    target="_blank"
  >
    Broken Robot
  </a>
  <svg
    fill="none"
    height="14"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="14"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
    />
    <polyline
      points="15 3 21 3 21 9"
    />
    <line
      x1="10"
      x2="21"
      y1="14"
      y2="3"
    />
  </svg>
</span>
`;
